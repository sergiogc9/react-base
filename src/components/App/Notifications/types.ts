import { Notification, QueueNotification } from '@src/types/notification';

export type StateProps = {
	notifications: QueueNotification[]
};

export type DispatchProps = {
	onAddNotification: (notification: Notification) => void,
	onUnqueueNotification: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps;
