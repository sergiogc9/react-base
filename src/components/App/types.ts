import { TenantObject } from "@src/class/Tenant";
import { UserObject } from "@src/class/User";
import { FacebookAuthStatus } from "@src/types/facebook";
import { TProps } from '@src/lib/i18n';

export type StateProps = {
	authenticated: boolean
	tenant: TenantObject | null
	user: UserObject | null
};

export type DispatchProps = {
	onAuth: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
