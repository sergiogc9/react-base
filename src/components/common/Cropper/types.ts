type Props = {
	/**
	 * The image in Base64. If null, no image has been set before.
	 */
	readonly image: string;
	/**
	 * A callback method called with the cropped image data
	 */
	readonly onImageCropped: (image: string) => void;
};

export type CropperProps = Props;
