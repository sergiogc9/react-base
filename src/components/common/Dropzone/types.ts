import { TProps } from '@src/lib/i18n';

export type AddFileError = {
	error: 'file_too_large' | 'incorrect_file_type'
	file: File
};

export type StateProps = {};

export type DispatchProps = {};

export type OwnProps = {
	files: File[]
	multiple: boolean
	accept?: string
	maxSize?: number
	showError?: boolean
	onFilesAdded(files: File[]): void
	onAddFilesError(errors: AddFileError[]): void
	onRemoveFile(file: File): void
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
