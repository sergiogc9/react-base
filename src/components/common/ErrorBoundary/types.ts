export type ErrorBoundaryProps = React.PropsWithChildren;
