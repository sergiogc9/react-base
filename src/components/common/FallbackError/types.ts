import { FallbackProps } from 'react-error-boundary';

export type FallbackErrorProps = FallbackProps;
