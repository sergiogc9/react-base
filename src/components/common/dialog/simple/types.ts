import { TProps } from '@src/lib/i18n';

export type StateProps = {};

export type DispatchProps = {};

export type OwnProps = {
	id: string
	text: string
	content: JSX.Element
	onAccept?: () => void
	onAcceptText?: string
	onCancel?: () => void
	onCancelText?: string
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
