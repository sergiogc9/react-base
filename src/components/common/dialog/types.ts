import { TProps } from '@src/lib/i18n';

export type StateProps = {
};

export type DispatchProps = {
};

export type OwnProps = {
	id: string
	icon?: JSX.Element
	title: string
	content: JSX.Element
	onAccept?: () => void
	onAcceptText?: string
	onCancel?: () => void
	onCancelText?: string
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
