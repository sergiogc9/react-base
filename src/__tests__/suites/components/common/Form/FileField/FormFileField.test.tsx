import React from "react";
import { act } from 'react-dom/test-utils';

import FormFileField from "@src/components/common/Form/FileField/";
import FormFileFieldWithoutStore from "@src/components/common/Form/FileField/FormFileField";
import { ComponentProps } from "@src/components/common/Form/FileField/types";
import { shallow } from "enzyme";
import TestHelper from "@src/__tests__/utils/Helper/TestHelper";

const element = TestHelper.fileFieldelement;

const file = new File([], "");

let onChangeMock = jest.fn();
let onFileErrorsMock = jest.fn();
let onAddNotificationMock = jest.fn();
describe("File field common component", () => {
	let wrapper: any;

	const updateComponent = (props: Partial<ComponentProps> = {}) => {
		act(() => {
			wrapper = TestHelper.getWrappedComponent(
				<FormFileField
					element={element}
					error={false}
					onChangeFile={onChangeMock}
					{...props}
				/>, {}).component;
		});
	};

	const getComponent = (props: Partial<ComponentProps> = {}) => (
		<FormFileFieldWithoutStore
			element={element}
			error={false}
			onChangeFile={onChangeMock}
			onAddNotification={onAddNotificationMock}
			{...props}
		/>
	);

	beforeEach(() => {
		onChangeMock = jest.fn();
		onFileErrorsMock = jest.fn();
		onAddNotificationMock = jest.fn();
		updateComponent();
	});

	it("Component default snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component default snapshot with multiple set to true", () => {
		updateComponent({ element: { ...element, multiple: true } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component snapshot on remove file", () => {
		act(() => wrapper.find('Dropzone').at(0).props().onRemoveFile());
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Component on added files", () => {
		act(() => wrapper.find('Dropzone').at(0).props().onFilesAdded([file]));
		expect(onChangeMock).toHaveBeenCalledTimes(1);
	});

	it("Component on file too large error", () => {
		wrapper = shallow(getComponent());
		act(() => wrapper.find('Dropzone').at(0).props().onAddFilesError([{ error: "file_too_large" }]));
		expect(onAddNotificationMock).toHaveBeenCalledTimes(1);
	});

	it("Component on incorrect type error", () => {
		wrapper = shallow(getComponent());
		act(() => wrapper.find('Dropzone').at(0).props().onAddFilesError([{ error: "incorrect_file_type" }]));
		expect(onAddNotificationMock).toHaveBeenCalledTimes(1);
	});

	it("Component on other error", () => {
		wrapper = shallow(getComponent());
		act(() => wrapper.find('Dropzone').at(0).props().onAddFilesError([{ error: "other_error" }]));
		expect(onAddNotificationMock).toHaveBeenCalledTimes(1);
	});

	it("Component on file errors handle", () => {
		updateComponent({ element: { ...element, onFileErrors: onFileErrorsMock } });
		act(() => wrapper.find('Dropzone').at(0).props().onAddFilesError([{ error: "other_error" }]));
		expect(onFileErrorsMock).toHaveBeenCalledTimes(1);
	});

	it("Component on validate base validation", () => {
		updateComponent({ element: { ...element, validations: { fn: () => false } } });
		act(() => wrapper.find('Dropzone').at(0).props().onFilesAdded([file]));
		expect(onChangeMock).toHaveBeenCalledWith(element.id, [file], false);
	});

	it("Component on validate correct", () => {
		updateComponent({ element: { ...element, validations: {} } });
		act(() => wrapper.find('Dropzone').at(0).props().onFilesAdded([file]));
		expect(onChangeMock).toHaveBeenCalledWith(element.id, [file], true);
	});

	it("Component snapshot after on force update", () => {
		updateComponent({ forceValue: [file] });
		expect(onChangeMock).toHaveBeenCalledWith(element.id, [file], true);
		expect(wrapper.html()).toMatchSnapshot();
	});
});
