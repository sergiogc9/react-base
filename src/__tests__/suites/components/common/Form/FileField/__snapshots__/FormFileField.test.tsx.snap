// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File field common component Component default snapshot 1`] = `"<div class=\\"discover-form-file-field\\"><div id=\\"dropzoneContainer\\" class=\\"dropzone-outer-container\\"><div class=\\"dropzone-inner-container\\"><div class=\\"dropzone-buttons\\"><div class=\\"md-inline-block md-file-input-container\\"><label for=\\"dropzoneFileInput\\" role=\\"button\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-btn md-btn--raised md-btn--text md-pointer--hover md-text\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><span class=\\"md-icon-text\\">newsletter.digest.edit.custom_image.upload.addFiles.one</span></div></label><input id=\\"dropzoneFileInput\\" type=\\"file\\" aria-hidden=\\"true\\" class=\\"md-file-input\\" tabindex=\\"-1\\" value=\\"\\"></div><div class=\\"dropzone-help\\"><i class=\\"md-icon material-icons icon-upload\\"></i><span>newsletter.digest.edit.custom_image.upload.dropFiles.one</span></div></div></div></div></div>"`;

exports[`File field common component Component default snapshot with multiple set to true 1`] = `"<div class=\\"discover-form-file-field\\"><div id=\\"dropzoneContainer\\" class=\\"dropzone-outer-container\\"><div class=\\"dropzone-inner-container\\"><div class=\\"dropzone-buttons\\"><div class=\\"md-inline-block md-file-input-container\\"><label for=\\"dropzoneFileInput\\" role=\\"button\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-btn md-btn--raised md-btn--text md-pointer--hover md-text\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><span class=\\"md-icon-text\\">newsletter.digest.edit.custom_image.upload.addFiles.one</span></div></label><input id=\\"dropzoneFileInput\\" type=\\"file\\" multiple=\\"\\" aria-hidden=\\"true\\" class=\\"md-file-input\\" tabindex=\\"-1\\" value=\\"\\"></div><div class=\\"dropzone-help\\"><i class=\\"md-icon material-icons icon-upload\\"></i><span>newsletter.digest.edit.custom_image.upload.dropFiles.many</span></div></div></div></div></div>"`;

exports[`File field common component Component snapshot after on force update 1`] = `"<div class=\\"discover-form-file-field\\"><div id=\\"dropzoneContainer\\" class=\\"dropzone-outer-container filled-container\\"><div class=\\"dropzone-inner-container\\"><div class=\\"dropzone-buttons\\"><div class=\\"md-inline-block md-file-input-container\\"><label for=\\"dropzoneFileInput\\" role=\\"button\\" class=\\"md-fake-btn md-fake-btn--no-outline md-btn md-btn--flat md-btn--text md-btn--raised-disabled md-text--disabled\\" aria-pressed=\\"false\\" disabled=\\"\\"><div class=\\"md-icon-separator\\"><span class=\\"md-icon-text\\">newsletter.digest.edit.custom_image.upload.addFiles.one</span></div></label><input id=\\"dropzoneFileInput\\" type=\\"file\\" aria-hidden=\\"true\\" class=\\"md-file-input\\" tabindex=\\"-1\\" value=\\"\\" disabled=\\"\\"></div><div class=\\"dropzone-help\\"><i class=\\"md-icon material-icons icon-upload\\"></i><span>newsletter.digest.edit.custom_image.upload.dropFiles.one</span></div></div><div id=\\"dropzoneUploadedFiles\\"><div class=\\"dropzone-upload-files-inner\\"><div class=\\"dropzone-uploaded-file\\"><div class=\\"dropzone-uploaded-file-info\\"><div class=\\"dropzone-uploaded-file-name\\"></div></div><div class=\\"dropzone-uploaded-file-commands\\"><button id=\\"dropzoneUploadedFile0RemoveButton\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block dropzone-uploaded-file-commands-remove\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">close</i><span class=\\"md-icon-text\\"></span></div></button></div></div></div></div></div></div></div>"`;

exports[`File field common component Component snapshot on remove file 1`] = `"<div class=\\"discover-form-file-field\\"><div id=\\"dropzoneContainer\\" class=\\"dropzone-outer-container\\"><div class=\\"dropzone-inner-container\\"><div class=\\"dropzone-buttons\\"><div class=\\"md-inline-block md-file-input-container\\"><label for=\\"dropzoneFileInput\\" role=\\"button\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-btn md-btn--raised md-btn--text md-pointer--hover md-text\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><span class=\\"md-icon-text\\">newsletter.digest.edit.custom_image.upload.addFiles.one</span></div></label><input id=\\"dropzoneFileInput\\" type=\\"file\\" aria-hidden=\\"true\\" class=\\"md-file-input\\" tabindex=\\"-1\\" value=\\"\\"></div><div class=\\"dropzone-help\\"><i class=\\"md-icon material-icons icon-upload\\"></i><span>newsletter.digest.edit.custom_image.upload.dropFiles.one</span></div></div></div></div></div>"`;
